default/python/AI/ShipDesignAI.py

```diff
@@ -2307,26 +2093,23 @@ def _get_design_by_name(design_name, update_invalid=False, looking_for_new_desig
     return design
 
 
-def get_part_type(partname):
+@cache_for_session
+def get_part_type(part_name):
     """Return the partType object (fo.getPartType(partname)) of the given partname.
 
     As the function in late game may be called some thousand times, the results are cached.
 
-    :type partname: str
+    :type part_name: str
```

Comment:
I think we don't need type strings anymore.   If you change it just use annotations.   Optional for this PR.